src/core/bittorrent/session.cpp

```diff
@@ -2111,6 +2122,9 @@ void Session::handleAddTorrentAlert(libtorrent::add_torrent_alert *p)
             }
         }
 
+        if (pref->isAddTrackersEnabled() && !torrent->isPrivate())
+            torrent->addTrackers(m_additionalTrackers);
```

Comment:
Okay after closer inspection: The magnets are just parsed add_torrent_params for us. So you can easily append the default trackers to each magnet's trackers before adding.
Something like:
```c++
//current code
if (fromMagnetUri) {
        p = magnetUri.addTorrentParams();
        hash = magnetUri.hash();
    }

// new code
if (fromMagnetUri) {
        p = magnetUri.addTorrentParams();
        p.trackers += m_additionalTrackers; //assuming std::vector here.
        hash = magnetUri.hash();
    }
```